src/main/java/org/folio/event/handler/EdiExportOrdersHistoryAsyncRecordHandler.java

```diff
@@ -45,42 +43,44 @@ public Future<String> handle(KafkaConsumerRecord<String, String> kafkaRecord) {
-                                .orElseThrow(() -> new IllegalStateException(TENANT_NOT_SPECIFIED_MSG));
-      Map<String, String> okapiHeaders = Map.of(OKAPI_HEADER_TENANT, tenantId);
-      exportHistoryService.createExportHistory(exportHistory, new DBClient(getVertx(), tenantId))
-        .compose(createdExportHistory -> {
-           return poLinesService.getPoLinesByLineIds(exportHistory.getExportedPoLineIds(), getContext(), okapiHeaders)
-                      .map(poLines -> updatePoLinesWithExportHistoryData(exportHistory, poLines))
-                      .compose(poLines -> {
-                        if (CollectionUtils.isNotEmpty(poLines)) {
-                          log.info("poLines not empty, updating them");
-                          return poLinesService.updatePoLines(poLines, new DBClient(getVertx(), tenantId));
-                        }
-                        log.info("Export EDI date was not updated : {}", createdExportHistory.getId());
-                        return Future.succeededFuture(0);
-                      })
-                      .map(updatedLines -> createdExportHistory);
-        })
-        .onComplete(ar -> {
-          if (ar.failed()) {
-            if (log.isErrorEnabled())
-              log.error("Can't store export history, kafkaRecord={}", kafkaRecord.value(), ar.cause());
-            promise.fail(ar.cause());
-          } else {
-            log.debug("Completed EdiExportOrdersHistoryAsyncRecordHandler.handle");
-            promise.complete(kafkaRecord.value());
-          }
-        });
-      return promise.future();
+        .orElseThrow(() -> new IllegalStateException(TENANT_NOT_SPECIFIED_MSG));
+      DBClient client = new DBClient(getVertx(), tenantId);
+      return exportHistory(exportHistory, client)
+        .map(v -> kafkaRecord.value());
     } catch (Exception e) {
       log.error("Failed to process export history kafka record, kafkaRecord={}", kafkaRecord, e);
       return Future.failedFuture(e);
     }
   }
 
+  Future<Void> exportHistory(ExportHistory exportHistory, DBClient client) {
+    return exportHistoryService.createExportHistory(exportHistory, client)
+      .compose(createdExportHistory -> client.getPgClient().withConn(conn ->
+        poLinesService.getPoLinesByLineIdsByChunks(exportHistory.getExportedPoLineIds(), conn)
+          .map(poLines -> updatePoLinesWithExportHistoryData(exportHistory, poLines))
+          .compose(poLines -> {
+            if (CollectionUtils.isNotEmpty(poLines)) {
+              log.info("poLines not empty, updating them");
+              return poLinesService.updatePoLines(poLines, conn, client.getTenantId());
+            }
+            log.info("Export EDI date was not updated : {}", createdExportHistory.getId());
+            return Future.succeededFuture(0);
+          })
+      ))
+      .onComplete(ar -> {
+        if (ar.failed()) {
+          if (log.isErrorEnabled())
```

Comment:
log.error with Supplier arguments could be used, there would not be need for log level check in this case.

Reply:
At least in this case, because the code is using both parameters and a throwable, I think it is more readable with `isErrorEnabled()`. Otherwise a cast is needed.